### PR TITLE
feat: implement onboarding modal

### PR DESCRIPTION
**Pull request by okanisildar**
Not sure about the component name 😬 

CleanShot 2023-12-14 at 23 53 47@2x

For testing purposes I've rendered inside of the `<RootModal>` in episode/:id page

CleanShot 2023-12-14 at 23 55 23@2x

### CODE REVIEW

**Comment by okanisildar** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #55** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/55?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/experiment-station/beecast/55?utm_source=stack-comment).

File: components/onboarding-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Button, DialogClose, Flex, Heading, Text } from '@radix-ui/themes';
+
+import { Logo } from './ui/logo';
+
+export function OnboardingModal() {
+  return (
+    <Flex direction="column" gap="2">
+      <Flex align="center" direction="row" gap="1">
```
Comment (altaywtf) [inline components/onboarding-modal.tsx:8]: **nit**: a little bit more spacing

```suggestion
      <Flex align="center" direction="row" gap="2">
```

**Review by altaywtf — APPROVED** [review]

File: lib/services/spotify/save-user-info.ts [hunk 2]
```diff
@@ -71,4 +71,6 @@ export const saveUserInfo = async ({
   if (isNewAccount) {
     await notifySlack(`🐝 New sign-up for *beecast*: ${data.display_name}`);
   }
+
+  return true;
```
Comment (altaywtf) [inline lib/services/spotify/save-user-info.ts:75]:
```suggestion
  return { isNewAccount } ;
```

File: app/(auth)/auth/callback/route.ts [hunk 3]
```diff
@@ -34,7 +34,16 @@ export async function GET(request: NextRequest) {
   }
 
   try {
-    await saveUserInfo({ session, user });
+    const isNewAccount = await saveUserInfo({ session, user });
```
Comment (altaywtf) [inline app/(auth)/auth/callback/route.ts:37]:
```suggestion
    const { isNewAccount } = await saveUserInfo({ session, user });
```

File: app/onboarding/start/page.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import Link from 'next/link';
+
+export default function Page() {
+  return (
+    <Flex direction="row" style={{ justifyContent: 'center' }}>
+      <Flex
+        direction="column"
+        gap="2"
+        style={{
+          width: 580,
```
Comment (altaywtf) [inline app/onboarding/start/page.tsx:12]:
```suggestion
          width: 'var(--container-1)',
```

File: app/onboarding/start/page.tsx
Comment (altaywtf) [inline app/onboarding/start/page.tsx:1]: **suggestion:** we may consider creating a component with an API like this:

- `Onboarding.Root`: wrapper
- `Onboarding.Title`: logo + title
- `Onboarding.Content`: body text
- `Onboarding.Actions`: flex row
- `Onboarding.Confirm`: confirm button
- `Onboarding.Cancel`: cancel button

or 

- `Onboarding.Root`: wrapper
- `Onboarding.Content`: title + content
- `Onboarding.Actions`: flex row
- copy pasta confirm / cancel buttons


and re-use them 🤔

Comment (altaywtf) [reply app/onboarding/start/page.tsx:1]: or just get rid of modals and intercepting routes, we found out how they work 😁

File: lib/services/spotify/save-user-info.ts (hunk 2 above)
Comment (okanisildar) [reply lib/services/spotify/save-user-info.ts:75]: lol I forgot to fix this return true placement before commit :D

**Review by altaywtf — APPROVED** [review]

**Pull request merged**